### PR TITLE
GSSoC'24: Tiktok Hand-Gesture Clone

### LINKED ISSUE #1048: GSSoC'24: Clone of tiktok's hand gesture filter
### Is your feature request related to a problem? Please describe.

I would like to add a clone of tiktok filter which uses hand gesture to recognize it, and take the image of the frame.

### Describe the solution you'd like.

I would like to add the filter with OpenCV and Mediapipe. The former provides the base of operation while the latter is used to get the orientation and the keypoints of the hand landmarks. Then, these landmarks will be fed to a neural network which will guess the gesture made by these keypoints. Upon the detection of the any of the three gesture, an image of the frame would be taken and will be included in the top where the gesture outlines would be there.

Tensorflow and Keras for training the neural network model.

### Describe alternatives you've considered.

For starters, a linear NN. If the initial model fails to provide satisfactory results, I would like to use machine learning algorithms like decision trees, or SVMs.

### Add any other context or screenshots about the feature request here.

For reference, the [link](https://www.youtube.com/watch?v=XAI-WszJFLg&ab_channel=FrigginKabin) for the video which shows the filter as a tiktok compilation.

### PR DESCRIPTION
**Pull request by SAM-DEV007**
### Have you read the [Contributing Guidelines](https://github.com/Kushal997-das/Project-Guidance/blob/main/CONTRIBUTING.md) ?

Yes, I have read the guidelines and adhered to following it.

### Description

The clone of Hand Gesture filter by Tiktok is added to the `AI/Intermediate/Hand_Gesture`. Two models are being used simultaneously, one is pretrained provided by `mediapipe` to detect hand landmarks and the other is trained to categorize the landmark data to one of the gestures.

The model training script and the training dataset is included along with the project to enable users to train their own models with varying inputs, or customizations.

It is also to be noted that the dataset is collected manually.

The project uses `OpenCV` to provide a GUI for the camera, along with the frame handling and the model detection from it. Whenever the user makes one of the gesture already mentioned on top of the screen, an image would be taken and will be resized to overlay on top of the detected gesture.

The images and detections are temporary, and none of them are cached or saved anywhere.

Fixes #1048 

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

None

### CODE REVIEW

**Review by Kushal997-das — CHANGES_REQUESTED** [review]
@SAM-DEV007 

1. [this](https://github.com/SAM-DEV007/Project-Guidance/blob/Tiktok-HandGesture-Clone/Artificial%20Intelligence/Intermediate/README.md) file should be under your project folder.
2. Describe about your project with good images and instructions.
3. Structured your folder correctly.

**Comment by SAM-DEV007** [comment]
@Kushal997-das I have done the changes and updated the forked branch. Let me know if any more changes are required.

**Review by Kushal997-das — APPROVED** [review]
👍🏻

**Comment by SAM-DEV007** [comment]
@Kushal997-das Can you possibly rename the `Level2` label to `level2` (lowercase L) label? The PR is shown but the points is not getting allocated in the leaderboard. I have seen merged PRs of others' as well and they have the same situation.

**Comment by Kushal997-das** [comment]
@SAM-DEV007 Done, do lemme know if it works or not.

**Comment by SAM-DEV007** [comment]
@SAM-DEV007 Yes, it did work! Sorry for the late response, I was waiting for the leaderboard to refresh.
